src/lib/user-state/userSessionStore.ts

```diff
@@ -0,0 +1,39 @@
+import { invalidate } from "$app/navigation";
+import type { User } from "$lib/model/api/user";
+
+export const USER_SESSION_DEPENDENCY_KEY = "data:getCurrentUser";
+
+// Internal cache for the user object.
+// undefined: State is unknown, never fetched, or cache explicitly invalidated
+// null: Fetch attempted, but user is confirmed not authenticated, or an error implies no user
+// User: Fetch successful, user is authenticated
+let SVELTE_KIT_INTERNAL_cachedUser: User | null | undefined = undefined;
+
+/**
+ * Retrieves the internal cached user object.
+ * This is primarily for use by the root layout load function.
+ *
+ * @returns The ached user object, or null if the user is not authenticated.
+ *          Returns undefined if the user state is unknown (e.g., not yet fetched or cache invalidated).
+ */
+export function getCachedUser(): User | null | undefined {
+    return SVELTE_KIT_INTERNAL_cachedUser;
+}
+
+/**
+ * Updates the internal cached user cache.
+ * This is primarily for use by the root layout load function after a successful fetch or auth check.
+ *
+ * @param user - The user object to cache, or null if the user is not authenticated.
+ */
+export function setCachedUser(user: User | null): void {
+    SVELTE_KIT_INTERNAL_cachedUser = user;
+}
+
+/**
+ * Invalidate the client-side user cache and trigger a re-fetch of the user session in the root layout.
```

Comment:
```suggestion
 * Invalidates the client-side user cache and trigger a re-fetch of the user session in the root layout.
```
Maybe user sessions is not the best wording as it could be confusing with normal login sessions?

Reply:
You're right. Session is not the best wording. It only caches the current user, so I try to reflect that.

Reply:
In my opinion you improved the wording a lot, however, you still are inconsistent with _Invalidate_ and _Retrieve**s**_ at the beginning of your comments.